src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -6,44 +6,138 @@
     {
-        public HashSet<string> Tags
-        {
-            get { return TagsObj.Count > 0 ? TagsObj : null; }
-            set { TagsObj = value ?? new HashSet<string>(); }
-        }
+        /// <summary>
+        /// Token Id.
+        /// Previously known as DefinitionId.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Represents the type of a structured token.
+        /// </summary>
+        public StructuredTokenKind Kind { get; set; } = StructuredTokenKind.Content;
+        
+        /// <summary>
+        /// The token value which will be displayed. Spacing tokens don't need to have value.
+        /// </summary>
+        public string Value { get; set; } = string.Empty;
+        
+        /// <summary>
+        /// Properties of the token.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>GroupId: `doc` to group consecutive comment tokens as documentation.</description>
+        /// </item>
+        /// <item>
+        /// <description>NavigateToId: Id for navigating to where the node where the token is defined. Should match the definitionId of the referenced node.</description>
+        /// </item>
+        /// </list>
+        /// </summary>
         public Dictionary<string, string> Properties
         {
             get { return PropertiesObj.Count > 0 ? PropertiesObj : null; }
             set { PropertiesObj = value ?? new Dictionary<string, string>(); }
         }
+        
+        /// <summary>
+        /// Add css classes for how the tokens will be rendered.
+        /// Each value needs a corresponding css for the class.
+        /// Contact APIView team for new classes.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>comment</description>
+        /// </item>
+        /// <item>
+        /// <description>keyword</description>
+        /// </item>
+        /// <item>
+        /// <description>literal</description>
+        /// </item>
+        /// <item>
+        /// <description>mname</description>
+        /// </item>
+        /// <item>
+        /// <description>punc</description>
+        /// </item>
+        /// <item>
+        /// <description>sliteral</description>
+        /// </item>
+        /// <item>
+        /// <description>text</description>
+        /// </item>
+        /// <item>
+        /// <description>tname</description>
+        /// </item>
+        /// </list>
+        /// </summary>
         public HashSet<string> RenderClasses
         {
             get { return RenderClassesObj.Count > 0 ? RenderClassesObj : null; }
             set { RenderClassesObj = value ?? new HashSet<string>(); }
         }
-        public string Value { get; set; } = string.Empty;
-        public string Id { get; set; }
-        public StructuredTokenKind Kind { get; set; } = StructuredTokenKind.Content;
+
+        /// <summary>
+        /// Opt in or opt out boolean properties
```

Comment:
You could say it that way. But my thinking is a little different. Some tags are opt in e.g. `Deprecated` since by default things are not deprecated. Some are opt out `HideFromNav` since by default nodes are show in navigation. Whatever makes it clearer is fine.